internal/output/html.go

```diff
@@ -0,0 +1,486 @@
+package output
+
+import (
+	"embed"
+	"fmt"
+	"html/template"
+	"io"
+	"slices"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/google/osv-scanner/internal/identifiers"
+	"github.com/google/osv-scanner/internal/semantic"
+	"github.com/google/osv-scanner/internal/utility/severity"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+type HTMLResult struct {
+	HTMLVulnCount    HTMLVulnCount
+	EcosystemResults []EcosystemResult
+}
+
+type EcosystemResult struct {
+	Ecosystem string
+	Sources   []SourceResult
+}
+
+type SourceResult struct {
+	Source         string
+	Ecosystem      string
+	PackageResults []PackageResult
+	PackageCount   [2]int // called and uncalled package count
+	HTMLVulnCount  HTMLVulnCount
+}
+
+type PackageResult struct {
+	Name             string
+	Ecosystem        string
+	Source           string
+	CalledVulns      []HTMLVulnResult
+	UncalledVulns    []HTMLVulnResult
+	InstalledVersion string
+	FixedVersion     string
+	HTMLVulnCount    HTMLVulnCount
+}
+
+type HTMLVulnResult struct {
+	Summary HTMLVulnResultSummary
+	Detail  map[string]string
+}
+
+type HTMLVulnResultSummary struct {
+	ID               string
+	PackageName      string
+	InstalledVersion string
+	FixedVersion     string
+	SeverityRating   string
+	SeverityScore    string
+}
+
+type HTMLVulnCount struct {
+	Critical int
+	High     int
+	Medium   int
+	Low      int
+	Unknown  int
+	Called   int
+	Uncalled int
+	Fixed    int
+	UnFixed  int
+}
+
+const UNFIXED = "No fix available"
+
+// HTML templates directory
+const TEMPLATEDIR = "html/*"
+
+// supportedBaseImages lists the supported OS base images for container scanning.
+var baseImages = []string{"Debian", "Alpine", "Ubuntu"}
+
+//go:embed html/*
+var templates embed.FS
+
+var vulnIndex = 0
```

Comment:
Suggest using an atomic integer here just in case we multi thread this in the future.

Reply:
And add a comment describing what this variable is for.

Reply:
Removed this variable and replaced with random method